solidity/contracts/bridge/Bridge.sol

```diff
@@ -1227,6 +1284,62 @@ contract Bridge is Ownable {
         bank.transferBalance(treasury, outputsInfo.totalTreasuryFee);
     }
 
+    /// @notice Submits a wallet fraud challenge.
+    /// @param walletPublicKey The public key of the wallet in the uncompressed
+    ///                        and unprefixed format (64 bytes).
+    /// @param walletPubKeyHash 20-byte public key hash (computed using Bitcoin
+    ///        HASH160 over the compressed ECDSA public key) of the wallet which
+    ///        performed the redemption transaction.
+    /// @param sighash Hash of the data (preimage) used for generating signature
+    ///                for the given input.
+    /// @param v Signature recovery value.
+    /// @param r Signature r value.
+    /// @param s Signature s value.
+    function submitFraudChallenge(
+        bytes memory walletPublicKey,
+        bytes20 walletPubKeyHash,
+        bytes32 sighash,
+        uint8 v,
+        bytes32 r,
+        bytes32 s
+    ) external payable {
+        // TODO: Consider calculating wallet public key hash instead of passing
+        // it as a parameter.
+        require(
+            wallets[walletPubKeyHash].state != WalletState.Closed &&
+                wallets[walletPubKeyHash].state != WalletState.Terminated,
+            "Wallet is closed or terminated"
+        );
```

Comment:
This allows making a challenge against an `Unknown` wallet though. I think it makes more sense to allow submitting challenges only for `Active` and `MovingFunds` wallets since only such wallets are supposed to make transactions on the BTC chain.

Reply:
Done.